lucene/facet/src/java/org/apache/lucene/facet/taxonomy/directory/DirectoryTaxonomyReader.java

```diff
@@ -322,9 +324,14 @@ public FacetLabel getPath(int ordinal) throws IOException {
         return res;
       }
     }
```

Comment:
Pre-existing: I don't like that we `return null` up above if the requested `ordinal` is out-of-bounds.  That's dangerous leniency and likely means the user is refreshing their main `IndexReader` and the `TaxonomyReader` in the wrong order.  It would be better to throw an exception here?  @gautamworah96 could you open a follow-on issue to fix that?  Thanks.

Reply:
Opened [LUCENE-9460](https://issues.apache.org/jira/browse/LUCENE-9460?filter=-2) for this